src/main/java/co/rsk/federate/FederationProviderFromFederatorSupport.java

```diff
@@ -135,9 +139,54 @@ public Optional<Address> getRetiringFederationAddress() {
         return federatorSupport.getRetiringFederationAddress();
     }
 
+    @Override
+    public Optional<Federation> getProposedFederation() {
+        if (!federatorSupport.getConfigForBestBlock().isActive(RSKIP419)) {
+            return Optional.empty();
+        }
+       
+        Optional<Address> proposedFederationAddress = getProposedFederationAddress();
+        Integer federationSize = federatorSupport.getProposedFederationSize()
+            .orElse(FEDERATION_NON_EXISTENT.getCode());
```

Comment:
i think this should be being done in the bridge, [as we are doing with the retiring fed](https://github.com/rsksmart/rskj/blob/feature/powpeg_validation_protocol-phase4/rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java#L237)

wdyt? The fed support should just return what it gets from the bridge.

Reply:
This is also being done in the Bridge for the proposed fed. The fed support also only returns what it gets from the Bridge, here we are explicitly defining how to treat an empty optional in case we receive it. wdyt?